api/agent/lb_agent.go

```diff
@@ -177,10 +212,41 @@ func (a *lbAgent) Submit(callI Call) error {
 	statsDequeue(ctx)
 	statsStartRun(ctx)
 
-	err = a.placer.PlaceCall(a.rp, ctx, call)
+	if call.Type == models.TypeDetached {
+		return a.placeDetachCall(ctx, call)
+	}
+	return a.placeCall(ctx, call)
+}
+
+func (a *lbAgent) placeDetachCall(ctx context.Context, call *call) error {
+	errPlace := make(chan error, 1)
+	rw := call.w.(*DetachedResponseWriter)
+	go a.spawnPlaceCall(ctx, call, errPlace)
+	select {
+	case err := <-errPlace:
+		return err
+	case <-rw.acked:
+		return nil
+	}
+}
+
+func (a *lbAgent) placeCall(ctx context.Context, call *call) error {
+	err := a.placer.PlaceCall(a.rp, ctx, call, a.placer.Config().PlacerTimeout)
 	return a.handleCallEnd(ctx, call, err, true)
 }
 
+func (a *lbAgent) spawnPlaceCall(ctx context.Context, call *call, errCh chan error) {
+	var cancel func()
+	ctx = common.BackgroundContext(ctx)
+	placerTimeout := a.placer.Config().DetachedPlacerTimeout
+	// PlacerTimeout for Detached + call.Timeout (inside container)
```

Comment:
Here, I think the math needs to include docker-pull and other network/scheduler latencies. New upcoming changes to Fn will put upper bounds to this, so we should leave this headroom as a configurable, where ctx timeout needs to be placerTimeout + call.Timeout + headroom.

Reply:
Remember here we are simulating a connected client, but placer timeout is a specific shorter timeout inside placers (eg. 30 secs). This means, we can only tolerate 30 secs of searching for a runner. On the other hand, clients should be able to wait for potentially long docker-pull and other scheduler/network latencies in addition to placer timeout and call.Timeout.

Reply:
we can leave headroom up to client timeouts? placer timeout as well is pretty much just a client timeout for detached specifically? at least, wrt placer timeout seems like we can zoom out a little, shouldn't be thinking about call timeout even really should we? if a client sets a 2s timeout on their http request, doesn't do us a lot of good to set 30s call timeout to enclose that - they seem like separate things.